ui/static/ui/js/listing.js

```diff
@@ -1,6 +1,78 @@
 define(['jquery', 'bootstrap', 'icheck', 'csrf'], function(jQuery) {
   'use strict';
 
+  /* global $ */
+
+  function formatGroupName(string) {
+    string = string.charAt(0).toUpperCase() + string.slice(1);
+    return string.substring(0, string.length - 1);
+  }
+
+  /* This is going to grow up to check whether
+   * the name is a valid Kerberos account
+   */
+  function isMITEmail(kerberosId) {
+    var email = kerberosId + '@mit.edu';
+    var regex = /^([a-zA-Z0-9_.+-])+\@(([a-zA-Z0-9-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+    return regex.test(email);
+  }
+
+  function formatUserGroups(userList, dest) {
+    var $dest = $(dest);
+    for (var i = 0; i < userList.length; i++) {
+      $dest.append(
+        '<div class="row">\n' +
+          '<div class="col-sm-7">\n' +
+            '<div class="cd-panel-members-list-username">' +
+              userList[i].username + '@mit.edu</div>\n' +
+          '</div>\n' +
+          '<div class="col-sm-3">\n' +
+            '<div class="cd-panel-members-list-group_type">' +
+              formatGroupName(userList[i].group_type) + '</div>\n' +
+          '</div>\n' +
+          '<div class="col-sm-2">\n' +
+            '<button ' +
+               'class="btn btn-default cd-panel-members-remove" ' +
+               'data-username="' + userList[i].username + '" ' +
+               'data-group_type="' + userList[i].group_type + '">\n' +
+               '<i class="fa fa-minus"></i>\n' +
+            '</button>\n' +
+          '</div>\n' +
+        '</div>\n');
+    }
+  }
+
+  function resetUserGroupForm() {
+    $('input[name=\'members-username\']').val('');
+    $('select[name=\'members-group_type\']').prop('selectedIndex', 0);
+  }
+
+  function showMembersAlert(message, mtype) {
+    //default value for mtype
+    mtype = typeof mtype !== 'undefined' ? mtype : 'success';
+    //reset all the classes
+    $('#members-alert').html(
+      '<div class="alert alert-' + mtype +
```

Comment:
Again, might be better with _., but eventually react should do this, so it may be a moot point if we are going to change it later anyway. Up to you, but you should file an issue to do this all with react